drivers/sensor/adxl367/adxl367.c

```diff
@@ -0,0 +1,1000 @@
+/*
+ * Copyright (c) 2023 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl367
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <string.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys/__assert.h>
+#include <stdlib.h>
+#include <zephyr/logging/log.h>
+
+#include "adxl367.h"
+
+LOG_MODULE_REGISTER(ADXL367, CONFIG_SENSOR_LOG_LEVEL);
+
+static const uint8_t adxl367_scale_mul[3] = {1, 2, 4};
+static uint8_t samples_per_set;
+
+/**
+ * @brief Configures activity detection.
+ *
+ * @param dev         - The device structure.
+ * @param th          - Structure holding the activity threshold information:
+ *				Enable/Disable Activity detection
+ *				Enable/Disable Referenced Activity detection
+ *				Set Threshold value
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl367_setup_activity_detection(const struct device *dev,
+					    const struct adxl367_activity_threshold *th)
+{
+	struct adxl367_data *data = dev->data;
+	int ret;
+
+
+	ret = data->hw_tf->write_reg_mask(dev, ADXL367_ACT_INACT_CTL,
+					  ADXL367_ACT_INACT_CTL_ACT_EN_MSK |
+					  ADXL367_ACT_INACT_CTL_ACT_REF_MSK,
+					  FIELD_PREP(ADXL367_ACT_INACT_CTL_ACT_EN_MSK, th->enable) |
+					  FIELD_PREP(ADXL367_ACT_INACT_CTL_ACT_REF_MSK,
+						     th->referenced));
+	if (ret)
+		return ret;
```

Comment:
Need braces even for single line blocks. See [Coding Style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style)